web/src/localization/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { FluentBundle, FluentResource } from '@fluent/bundle';
+import { mapBundleSync } from '@fluent/sequence';
+import { negotiateLanguages } from '@fluent/langneg';
+import { Accessor, JSX, createContext, createMemo, useContext } from 'solid-js';
+import { createLocalStorageSignal } from '../utils/localstorage';
+import { Option } from '../types';
+
+const ftls = import.meta.glob('./locale/*.ftl', { eager: true, as: 'raw' });
+const bundles = Object.fromEntries(
+  Object.entries(ftls).map(([fileName, ftl]) => {
+    const languageName = fileName.replace(
+      /^.*\/(.+)\..*$/,
+      (_, language: string) => language,
+    );
+    const resource = new FluentResource(ftl);
+    const bundle = new FluentBundle(languageName);
+    const errors = bundle.addResource(resource);
+    if (errors.length > 0) {
+      throw new Error(
+        `Failed to load translation bundle ${fileName} as ${languageName}`,
+        { cause: errors[0] },
+      );
+    }
+    return [languageName, bundle];
+  }),
+);
+
+const fallbackLanguage = 'en';
+const LanguageContext = createContext<
+  [
+    Accessor<string[]>,
+    {
+      setUserLanguage: (language: Option<string>) => void;
+      availableLanguages: () => { [language: string]: FluentBundle };
+    },
+  ]
+>();
+export const requireLanguageContext = () => {
+  const langContext = useContext(LanguageContext);
+  if (langContext == undefined) {
```

Comment:
This should be sufficient:

 ```suggestion
  if (!langContext) {
```

Reply:
That's true. I think the current setup of ESLint won't force you to explicitly compare against `undefined`. If we want to explicitly use `undefined`, I would like us to use `===` instead of `==`.

Reply:
Sure, a432e0c47876029cafc36a0da3b421c8e4feffb9